verilog/tools/ls/verilog-language-server_test.cc

```diff
@@ -1423,6 +1430,51 @@ TEST_F(VerilogLanguageServerSymbolTableTest, CheckReferenceUnknownSymbol) {
   ASSERT_EQ(response_b["result"].size(), 0);
 }
 
+TEST_F(VerilogLanguageServerSymbolTableTest, HoverOverSymbol) {
+  absl::string_view filelist_content = "mod.v\n";
+  static constexpr absl::string_view  //
+      module_content(
+          R"(module mod(
+    input clk,
+    input reg [31:0] a,
+    input reg [31:0] b,
+    output reg [31:0] sum);
+  always @(posedge clk) begin : addition
+    assign sum = a + b;
+  end
+endmodule
+)");
+
+  const verible::file::testing::ScopedTestFile filelist(
+      root_dir, filelist_content, "verible.filelist");
+  const verible::file::testing::ScopedTestFile module(root_dir, module_content,
+                                                      "mod.v");
+
+  const std::string module_open_request =
+      DidOpenRequest("file://" + module.filename(), module_content);
+  ASSERT_OK(SendRequest(module_open_request));
+
+  // obtain diagnostics for both files
+  GetResponse();
+
+  // find definition for "var1" variable of a module in b.sv file
+  std::string hover_request =
+      HoverRequest("file://" + module.filename(), 2, 6, 12);
+
+  ASSERT_OK(SendRequest(hover_request));
+  json response = json::parse(GetResponse());
+
+  // For now when the file is invalid we will not be able to obtain symbols
+  // from it if it was incorrect from the start
+  ASSERT_EQ(response["id"], 2);
+  ASSERT_EQ(response["result"].size(), 1);
+  ASSERT_EQ(response["result"]["contents"]["kind"], "markdown");
```

Comment:
could you cast  `response["result"]` to `verible::lsp::Hover`, so that we have a nice struct to inspect ? The sequence of nested `["..."]` accesses makes it harder to read.